src/main/kotlin/com/openlattice/chronicle/controllers/SurveyController.kt

```diff
@@ -0,0 +1,185 @@
+import com.openlattice.chronicle.util.ChronicleServerUtil
+import org.springframework.format.annotation.DateTimeFormat
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.*
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import javax.inject.Inject
+import javax.servlet.http.HttpServletResponse
+
+/**
+ * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
+ */
+
+@RestController
+@RequestMapping(CONTROLLER)
+class SurveyController @Inject constructor(
+    val surveysService: SurveysService,
+    val downloadService: DataDownloadService,
+    val idGenerationService: HazelcastIdGenerationService,
+    override val authorizationManager: AuthorizationManager,
+    override val auditingManager: AuditingManager
+) : SurveyApi, AuthorizingComponent {
+
+    @Timed
+    @GetMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + APP_USAGE_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getAppUsageSurveyData(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @RequestParam(value = START_DATE) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) startDateTime: OffsetDateTime,
+        @RequestParam(value = END_DATE) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) endDateTime: OffsetDateTime
+    ): List<AppUsage> {
+        return surveysService.getAppUsageData(studyId, participantId, startDateTime, endDateTime)
+    }
+
+    @Timed
+    @PostMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + APP_USAGE_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+    )
+    override fun submitAppUsageSurvey(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @RequestBody surveyResponses: List<AppUsage>
+    ) {
+        surveysService.submitAppUsageSurvey(studyId, participantId, surveyResponses)
+    }
+
+    @Timed
+    @PostMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun createQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @RequestBody questionnaire: Questionnaire
+    ): UUID {
+        ensureWriteAccess(AclKey(studyId))
+        val id = idGenerationService.getNextId()
+        surveysService.createQuestionnaire(studyId, id, questionnaire)
+
+        return id
+    }
+
+    @DeleteMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH]
+    )
+    override fun deleteQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID
+    ): OK {
+        ensureOwnerAccess(AclKey(studyId))
+        surveysService.deleteQuestionnaire(studyId, questionnaireId)
```

Comment:
This only deletes the questionnaire entity, right? What about the corresponding data, i.e. the responses/submissions?

Reply:
Yeah this only deletes the questionnaire. Deleting submissions should be part of participant data deletion workflow